source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/admin/invite-organization/display.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+
+import { tomorrow } from '@lib/date';
+
+import { OrganizationInviteAttributes } from '@data/models/organization-invite';
+
+export interface IProps {
+  onSubmit: (data: OrganizationInviteAttributes) => Promise<void>;
+}
+
+export interface IState {
+  name: string;
+  ownerEmail: string;
+  expiresAt: Date;
+}
+
+export default class InviteOrganizationDisplay extends React.Component<IProps, IState> {
+  state = { name: '', ownerEmail: '', expiresAt: tomorrow() };
+
+  submit = async (e: Event) => {
+    e.preventDefault();
+
+    await this.props.onSubmit(this.state);
+    this.reset();
+  }
+
+  reset = () => {
+    this.setState({ name: '', ownerEmail: '', expiresAt: tomorrow() });
+  }
+
+  ownerEmailChanged = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
This can be inline in the render method especially if is only going to be an update in the state

onChange={(e) => this.setState('ownerEmail',e.target.value)}

Reply:
I like to keep the template simple. I agree that having methods for every event handler is obnoxious, though.

I think the solution will ultimately be a helper method that a class can have added to it.

I'm thinking about doing something like this:

```
constructor(...) {
  // ...
  setupTemplateHelpers(this); // assigns some helper methods on `this`
}
//...
onChange={this.mut('ownerEmail')}
```


alternatively, I could try a class decorator that assigns the helpers to the prototype and see if there is an easy to to access those.

thoughts?